src/utils/validateIdentification.test.ts

```diff
@@ -0,0 +1,52 @@
+import { validateAndCleanId } from "./validateIdentification";
+import { EnvVersionError } from "../services/envVersion";
+
+describe("validate and clean IDs", () => {
+  it("should return cleaned NRIC if variables are valid", () => {
+    expect.assertions(1);
+    const idNRICValidation = "NRIC";
+    expect(validateAndCleanId("s0000001i", idNRICValidation)).toBe("S0000001I");
+  });
+
+  it("should return cleaned alphanumeric ID if variables are valid", () => {
+    expect.assertions(1);
+    const idRegexValidation = "REGEX";
+    const alphanumericRegex = "^[a-zA-Z0-9-_ ]+$";
+    expect(
+      validateAndCleanId("abcd_1234", idRegexValidation, alphanumericRegex)
+    ).toBe("ABCD_1234");
+  });
+
+  it("should return existing input as ID if validation is missing", () => {
+    expect.assertions(1);
+    expect(validateAndCleanId("abcd_1234")).toBe("abcd_1234");
```

Comment:
Can I suggest to send a warning to Sentry when this happens? Ideally we'll always have validation, so this will let us know if validation is missing (: What do you think?

Reply:
yes! will be ensuring `idValidation` is not undefined, and will throw EnvVersionError (which is caught in `CollectCustomerDetailsScreen` and send warning to sentry)